crates/forge/tests/integration/l1_handler_executor.rs

```diff
@@ -69,16 +67,35 @@ fn l1_handler_execute() {
                     selector!("panicking_l1_handler")
                 );
 
-                l1_handler.from_address = 0x123;
-                l1_handler.payload = array![].span();
-                match l1_handler.execute() {
+                match l1_handler.execute(0x123, array![].span()) {
                     Result::Ok(_) => panic_with_felt252('should have panicked'),
                     Result::Err(panic_data) => {
                         assert(*panic_data.at(0) == 'custom', 'Wrong 1st panic datum');
                         assert(*panic_data.at(1) == 'panic', 'Wrong 2nd panic datum');
                     },
                 }
             }
+
+            #[test]
+            fn l1_handler_function_missing() {
+                let calldata = array![0x123];
+
+                let contract = declare("l1_handler_executor").unwrap();
+                let (contract_address, _) = contract.deploy(@calldata).unwrap();
+
+
+                let mut l1_handler = L1HandlerTrait::new(
+                    contract_address,
+                    selector!("this_does_not_exist")
+                );
+
+                match l1_handler.execute(0x123, array![].span()){
+                    Result::Ok(_) => panic_with_felt252('should have panicked'),
+                    Result::Err(_) => {
+                        // Would be nice to assert the error here once it is be possible in cairo
```

Comment:
This is the subject to the `TODO` with the issue you created i think, i'd leave it in the comment here

Reply:
Hmm, I am not sure about this, the problem here is a lack of utilities in cairo to format a byte array with our addresses to assert the message. This case works as intended.

Reply:
Why can't you do that? E.g. this works
```rs
#[test]
#[should_panic(expected: "string")]
fn test() {
    assert!(1 == 2, "string");
}
```

Reply:
Ofc I can, I cannot build the string with contract addresses though.

Reply:
> This is the subject to the TODO with the issue you created i think, i'd leave it in the comment here

I have assumed this will be one of those issues we will close in 10 months. But maybe it is better to add one

Reply:
https://github.com/foundry-rs/starknet-foundry/issues/2114